nomos-services/mix/src/backends/libp2p.rs

```diff
@@ -182,6 +200,22 @@ impl MixSwarm {
                     tracing::error!("Failed to send incoming message to channel: {e}");
                 }
             }
+            SwarmEvent::Behaviour(nomos_mix_network::Event::EstabalishNewConnections {
+                amount,
+                excludes,
+            }) => {
+                tracing::debug!(
+                    "Establishing {amount} new connections excluding peers: {excludes:?}"
+                );
+                let successful_dials = self.dial_to_peers(amount, Some(&excludes));
+                if successful_dials < amount {
+                    tracing::warn!(
+                        "Tried to establish {} new connections, but only {} succeeded",
+                        amount,
+                        successful_dials
+                    );
+                }
+            }
```

Comment:
I think this is the wrong approach. Mix behaviour should not relay on external swarm to interpret what needs to be done. In this case I think it should generate the dialing events internally. Basically removing the `EstablishNewConnections` event and handle it internally in the behaviour itself.